managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/RangeCache.java

```diff
@@ -73,12 +73,11 @@ public RangeCache(Weighter<Value> weighter, TimestampExtractor<Value> timestampE
      * @return whether the entry was inserted in the cache
      */
     public boolean put(Key key, Value value) {
-        if (entries.putIfAbsent(key, value) == null) {
+        Value v = entries.computeIfAbsent(key, (k) -> {
             size.addAndGet(weighter.getSize(value));
-            return true;
-        } else {
-            return false;
-        }
+            return value;
+        });
+        return v == value;
```

Comment:
Nice catch! But look like it changed the original behavior. See the following test. Its success in the old code but failed in this change.
```java
    @Test
    public void test() {
        RangeCache<Integer, RefString> cache = new RangeCache<>();
        RefString s0 = new RefString("zero");
        assertEquals(s0.refCnt(), 1);
        assertTrue(cache.put(0, s0));
        assertFalse(cache.put(0, s0));
    }
```

Reply:
Got it, very thanks to you.